### ISSUE TITLE
Add API to allow writing PAM modules

### ISSUE THREAD

**Issue by Kilobyte22** [MrFloya/pam-auth#4]
Having the ability to authenticate against PAM is one thing. However you should also be able to implement your own pam modules using this API. Would be nice to have support for that.

**Comment by MrFloya** [comment]
Yeah I started with the client API because that was bit that I needed but I definitely plan to add support for modules. 
One Problem is I don't really know the APIs and data flows that well so I am not quite sure how to best design a ideomatic Rust API on top of them yet. It is on my backlog for sure.

**Comment by pixelou** [comment]
Hi just passing by and saw this issue. Maybe you can merge effort with @tozny for this (https://github.com/tozny/rust-pam)?

**Comment by carnott-snap** [comment]
Other options for support include [anonwell/pam-rs](https://github.com/anowell/pam-rs) and [rcatolino/pam_sm_rust](https://github.com/rcatolino/pam_sm_rust).

**Comment by 1wilkens** [comment]
Sorry for not responding earlier. I took a look at tozny/pam but figured it was unmaintained at the time. The other two options look promising and go into a direction I had also considered (using a macro to export module functions).

I want to add at least some basic support before 1.0 so this is definitely a priority. Once I have time, I'll contact the authors and see if we can work on adding module support to this library.

**Comment by carnott-snap** [comment]
Of the two, I have decided to integrate with rcatolino/pam_sm_rust, since it is on crates.io and has more features. Let me know if there is any help I can be integrating.
